react/features/device-selection/components/DeviceSelector.js

```diff
@@ -0,0 +1,185 @@
+import Select from '@atlaskit/single-select';
+import React, { Component } from 'react';
+
+import { translate } from '../../base/i18n';
+
+/**
+ * React component for selecting a device from a select element.
+ * Wraps Select with device selection specific logic.
+ *
+ * @extends Component
+ */
+class DeviceSelector extends Component {
+    /**
+     * DeviceSelector component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The list of devices to display in the select element.
+         */
+        devices: React.PropTypes.array,
+
+        /**
+         * If false, will return a selector with no selection options.
+         */
+        hasPermission: React.PropTypes.bool,
+
+        /**
+         * If true, will render the selector disabled with a default selection.
+         */
+        isDisabled: React.PropTypes.bool,
+
+        /**
+         * The Select component will adjust the top margin if this instance
+         * is the first of many.
+         */
+        isFirstChild: React.PropTypes.bool,
```

Comment:
By default, selectors will have 20px top margin if isFirstChild is falsy. I've made a change just now so this is hardcoded to true to prevent Selector from adding 20 top margin and using jitsi-meet css to adjust the margins between Selector instances.